src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
Wouldn't that make the assert less meaningful? Take this example:

```
#include <assert.h>

int main(int argc, char **argv)
{
    int i = 0;
    assert(i >= 1 && i <= -1 && i != 5 && i != 61);
    return 0;
}
```

It aborts with this message:
```
a.out: lel.c:6: main: Assertion `i >= 1 && i <= -1 && i != 5 && i != 61' failed.
```

If we had only one clause per assert statement, then the abort message would be more meaningful.

I'm not sure how many implementations of `assert.h` do this, but it can be implemented as a macro, so there's no function call overhead (although that argument might be redundant as it would probably be inlined anyways).